src/encode.cpp

```diff
@@ -253,4 +254,46 @@ operator>>(MessageBuffer& buffer, PublishIntentEnd& msg)
   return true;
 }
 
+void
+operator<<(messages::MessageBuffer& msg, const quicr::Name& val)
+{
+  constexpr uint8_t size = sizeof(quicr::Name::uint_type) * 2;
+  for (size_t i = 0; i < size; ++i)
+    msg << val[size - 1 - i];
+}
+
+bool
+operator>>(messages::MessageBuffer& msg, quicr::Name& val)
+{
+  constexpr uint8_t size = sizeof(quicr::Name::uint_type) * 2;
+  std::array<uint8_t, size> bytes;
+  for (int i = 0; i < size; ++i)
+    msg >> bytes[i];
+
+  val = Name{bytes.data(), size};
+
+  return true;
+}
+
+void
+operator<<(messages::MessageBuffer& msg, const quicr::Namespace& val)
+{
+  msg << val.length();
+  msg << val.name();
+}
+
+bool
+operator>>(messages::MessageBuffer& msg, quicr::Namespace& val)
+{
+  quicr::Name mask;
+  msg >> mask;
```

Comment:
mask seems like not the right variable name.  is it ```quicr::Name name_mask ```

Reply:
Could be. To me a variable of type Name that is called mask is implicitly understood to be a name mask, but there isn't any harm being more explicit. I'll add that.